### PR TITLE
eng-214

### PR DESCRIPTION
**Pull request by Connorbelez**
### TL;DR

Added comprehensive test suite for unified payment rails with 158 new tests across 7 test files, covering transfer state machine transitions, cash ledger mappings, webhook pipelines, and financial property invariants.

### What changed?

Added extensive test coverage for the unified payment rails system:

- **Transfer Machine Tests**: Complete coverage of XState transfer machine with all valid/invalid state transitions and action verification
- **Provider Registry Tests**: Unit tests for transfer provider resolution and mock provider environment gating
- **Cash Ledger Mapping Tests**: Verification that all 10 transfer types map to correct journal entry types and account families
- **Webhook Pipeline Tests**: End-to-end webhook simulation through MockTransferProvider to GT state transitions
- **Inbound Flow Integration Tests**: Full collection attempt to bridge transfer to obligation settlement pipeline
- **Outbound Flow Integration Tests**: Disbursement flows, failed transfer safety, and multi-leg deal closing scenarios
- **Financial Property Tests**: Deterministic tests verifying core money invariants (rounding conservation, exactly-once posting, replay safety, reversal completeness)

### How to test?

Run the new test suites:
```bash
# Run all new tests
bun test convex/payments/transfers/__tests__/

# Run specific test files
bun test convex/payments/transfers/__tests__/transferMachine.test.ts
bun test convex/payments/transfers/__tests__/cashLedgerMapping.test.ts
bun test convex/payments/transfers/__tests__/webhookPipeline.test.ts
bun test convex/payments/transfers/__tests__/inboundFlow.integration.test.ts
bun test convex/payments/transfers/__tests__/outboundFlow.integration.test.ts
bun test convex/payments/transfers/__tests__/financialProperties.test.ts
bun test convex/payments/transfers/providers/__tests__/registry.test.ts

# Verify zero regression
bun run test
```

### Why make this change?

This comprehensive test suite ensures zero regression when introducing unified payment rails by:

1. **Preventing State Machine Bugs**: Every valid and invalid transition is tested to catch state machine logic errors
2. **Ensuring Financial Safety**: Property tests verify critical money invariants like no double-posting and exact reversal cancellation
3. **Validating Integration Points**: End-to-end tests verify the D4 conditional correctly prevents duplicate cash postings between bridge and direct transfer paths
4. **Covering Edge Cases**: Tests scenarios like failed multi-leg transfers, webhook replays, and rounding edge cases that could cause financial discrepancies
5. **Maintaining Backward Compatibility**: All tests are additions only, ensuring existing functionality remains intact

The test suite provides confidence that the unified payment rails can be deployed without breaking existing payment flows or introducing financial inconsistencies.

## Release Notes

* **Tests**
  * Added comprehensive test coverage for payment transfer flows, including inbound collections, outbound disbursements, webhook processing, and transfer reversals.
  * Added financial property regression tests to verify payment rounding accuracy, ledger posting integrity, and idempotency across state transitions.

* **Code Quality**
  * Strengthened validation for webhook event types to prevent invalid data persistence.
  * Improved webhook event schema to enforce stricter type checking.

### CODE REVIEW

File: convex/payments/transfers/__tests__/inboundFlow.integration.test.ts [hunk 1]
```diff
@@ -0,0 +1,710 @@
...
+ *        non-bridged posts
+ * T-011: Full inbound flow — collection plan entry -> collection attempt ->
+ *        bridge transfer -> obligation PAYMENT_APPLIED
+ * T-014: Bridge idempotency — re-running emitPaymentReceived does not create
+ *        duplicate transfer
+ *
+ * ORDERING NOTE: T-012 and T-013 are placed BEFORE T-011/T-014 because
+ * finishAllScheduledFunctions (used by T-011/T-014) leaves the convex-test
+ * workflow component in a state where subsequent workflow.start calls fail.
+ * This is a known convex-test limitation — not a bug in application code.
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply convex/payments/transfers/__tests__/inboundFlow.integration.test.ts]: Fixed. Updated the ORDERING NOTE header comment to clarify that only T-011 uses `finishAllScheduledFunctions`. The comment now notes that T-014 does not use it but is placed before T-011 for the same ordering reason (convex-test workflow component state corruption).

File: convex/payments/transfers/__tests__/webhookPipeline.test.ts [hunk 2]
```diff
@@ -0,0 +1,582 @@
...
+			});
+			expect(dupActions).toEqual([]);
+
+			const [afterDupSnap] = transition(transferMachine, confirmedSnap, {
+				type: "FUNDS_SETTLED",
+				settledAt: 1_700_000_000_000,
+				providerData: {},
+			});
+			expect(afterDupSnap.value).toBe("confirmed");
+		});
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply convex/payments/transfers/__tests__/webhookPipeline.test.ts]: **Fix:** Agreed — T-009 was testing XState terminal-state idempotency (replaying events on a machine already in `confirmed`/`failed`/`reversed` produces zero transitions and zero actions), not actual webhook deduplication. The real dedup lives in `persistTransferWebhookEvent` which queries the `webhookEvents` table by `provider + providerEventId` index, and is already covered by integration tests in `eftVopayWebhook.test.ts`.

Renamed the T-009 suite from "Webhook deduplication" to "Terminal-state idempotency" across the describe block, sub-describes, and file header comment. Added a doc note pointing to where real webhook dedup coverage lives. Commit: eb5e1b2f
